clippy_lints/src/methods/mod.rs

```diff
@@ -756,6 +757,32 @@ declare_clippy_lint! {
     "using `Iterator::step_by(0)`, which will panic at runtime"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for the use of `iter.nth(0)`.
+    ///
+    /// **Why is this bad?** `iter.nth(0)` is unnecessary, and `iter.next()`
+    /// is more readable.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust,ignore
+    /// // Bad
+    ///  let mut s = HashSet::new();
+    ///  s.insert(1);
```

Comment:
can you add a `#` in front of those two lines, so they won't be shown in the documentation (since there not relevant for the prupose of the lint).

Also I don't see, why this doc test should be `ignore`.

Reply:
done